src/EventListener/AddToSearchIndexListener.php

```diff
@@ -48,6 +55,10 @@ public function onKernelTerminate(PostResponseEvent $event)
             return;
         }
 
+        if (preg_match('/' . preg_quote($this->fragmentPath, '/') .'/', $event->getRequest()->getPathInfo())) {
```

Comment:
Maybe that should be checked in `indexPageIfApplicable`, that's what the function's name says it does?